--- a/arch/x86/boot/tty.c	2016-08-14 21:37:59.568973337 +0800
+++ b/arch/x86/boot/tty.c	2016-08-14 21:38:26.645114908 +0800
@@ -118,7 +118,7 @@
 
 int getchar_timeout(void)
 {
-	int cnt = 30;
+	int cnt = 3;
 	int t0, t1;
 
 	t0 = gettime();
--- a/arch/x86/boot/video.c	2016-08-14 21:38:37.876173630 +0800
+++ b/arch/x86/boot/video.c	2016-08-14 21:39:33.645465227 +0800
@@ -205,7 +205,7 @@
 	unsigned int sel;
 
 	puts("Press <ENTER> to see video modes available, "
-	     "<SPACE> to continue, or wait 30 sec\n");
+	     "<SPACE> to continue, or wait 3 sec\n");
 
 	kbd_flush();
 	while (1) {
